### DISCUSSION TITLE
Use xarray to combine time dimension in a netcdf file

### DISCUSSION THREAD

**Discussion by SAMPHY86** [pydata/xarray#5374] [Q&A]
I have netcdf files in multi-dimension:

forecast_initial_time
forecast_hour
latitude
longitude
Variable
forecast_initial_time are 0600 UTC and 1800 UTC forecast_hour are from 1 to 12.

So, if forecast_initial_time = 0600 and forecast_hour = 2, it represent 0800 UTC.

I would like to convert (forecast_initial_time and forecast_hour) into time (0000 UTC to 2300 UTC). (which is normally follow in standard netcdf formate time, lat. lon)
Calculate daily average (which I am doing in xarray but need a deltatime function).
Calculate 12-hours average.
Thanks for any help.

File are https://drive.google.com/drive/folders/1v2oQ3aFOiqlBu7U78om1jQKeIAFXEGXq

**Comment by max-sixty** [comment]
Do you have an reproducible example that possible to copy paste?

**Reply by SAMPHY86** [reply]
I have a really basic code, which is not doing its job. The format can be checked in Panoply as well. Thanks a lot in advance. 

```python
import numpy as np
import xarray as xr
from datetime import datetime, timedelta
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib notebook

## Open the file
ds = xr.open_dataset('/home/Downloads/*.nc')

## Read the variable
ds = ds.MSDWSWRF

## Merge dimension 1 and 2 OR numpy reshape but it stack dimension 1 onto 2 which isn't required 
ds2 = xr.merge([ds])#, dim ="time")

### Save as netcdf with time dimension (datetime, lat, lon, MSDWSWRF)
ds2.to_netcdf(f'/home/Downloads/mergefile.nc')
```

**Reply by max-sixty** [reply]
I had meant copy & paste, including creating the inputs

**Reply by SAMPHY86** [reply]
No, I am sorry I don't have that. But input files can be downloaded/linked from the provided link.
